### DISCUSSION TITLE
Syntax

### DISCUSSION THREAD

**Discussion by null0ptr** [microsoft/vscode-cpptools#7655] [Q&A] [answered]
When I write an STL data structure such as vector, map using code completion IntelliSense, only the name gets completed not angular brackets. Also builtin functions like sort(), partition() doesn't place "()" after the function name. Do these problems have any solution? Kindly inform me if there any. Otherwise, I think solving these problems in upcoming updates would be a great idea.

**Comment by sean-mcmanus** [comment, marked as answer]
Completion of `<>` is not implemented, particularly because there are many cases in which the <> can be omitted, such as via
```
#include <vector>

int main()
{
    std::vector i = { 0 };
    return 0;
}
```

The () issue with std::sort/partition/etc. is fixed for our pending 1.5.0-insiders: https://github.com/microsoft/vscode-cpptools/issues/7626

**Reply by null0ptr** [reply]
Maybe giving an option for the "angular bracket case" will be a good idea. Like "use the angular bracket for STL data structure" as `vector<>` is the most used form.

**Reply by sean-mcmanus** [reply]
Can you file a feature request so others can upvote it?
